### ISSUE TITLE
EBS volumes not working as expected

### ISSUE THREAD

**Issue by madhurranjan** [kubernetes/kubernetes#21903]
I'm using 1.1.7 and have nodes spread across multiple zones.  I'm trying to bring up an influxdb cluster across 3 zones in a raft cluster and I'm trying to have pods per zone. I've faced a bunch of problems while trying to do this:

1. I created 3 volumes in 3 zones and added them as shown below with the file volume-list.yaml
```
kind: PersistentVolume
apiVersion: v1
metadata:
  name: influxvol1
  namespace: infra
  labels:
    name: influxvol
    zone: us-west-2a
spec:
  capacity:
    storage: 50Gi
  accessModes:
    - ReadWriteOnce
  persistentVolumeReclaimPolicy: Recycle
  awsElasticBlockStore:
     volumeID: aws://us-west-2a/vol-b98d464f
     fsType: ext4
---
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  namespace: infra
  name: influxclaim1
  labels:
     name: influxclaim1
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 50Gi
---
kind: PersistentVolume
apiVersion: v1
metadata:
  namespace: infra
  name: influxvol2
  labels:
    name: influxvol2
    zone: us-west-2b
spec:
  capacity:
    storage: 50Gi
  accessModes:
    - ReadWriteOnce
  persistentVolumeReclaimPolicy: Recycle
  awsElasticBlockStore:
     volumeID: aws://us-west-2b/vol-45431e85
     fsType: ext4
---
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  namespace: infra
  name: influxclaim2
  labels:
     name: influxclaim2
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 50Gi
---
kind: PersistentVolume
apiVersion: v1
metadata:
  namespace: infra
  name: influxvol3
  labels:
    name: influxvol3
    zone: us-west-2c
spec:
  capacity:
    storage: 50Gi
  accessModes:
    - ReadWriteOnce
  persistentVolumeReclaimPolicy: Recycle
  awsElasticBlockStore:
     volumeID: aws://us-west-2c/vol-a5f10a1c fsType: ext4
---
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  namespace: infra
  name: influxclaim3
  labels:
     name: influxclaim3
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 50Gi
```

Followed by that if I run the command , I can see 2 PVs associated with the same PVC.  
$ kubectl get pvc
NAME           LABELS              STATUS    VOLUME       CAPACITY   ACCESSMODES   AGE
influxclaim1   name=influxclaim1   Bound     influxvol1   50Gi       RWO           6m
influxclaim2   name=influxclaim2   Bound     influxvol3   50Gi       RWO           6m
influxclaim3   name=influxclaim3   Bound     influxvol3   50Gi       RWO           6m

2. I'd like to mention selector on PVC so that I can create PVCs per zone and when I add new PV's it can have labels based on Zones since EBS volumes are per zone. I'm not sure how this can be done. Tried the selector in PVC and its currently not in the PVC spec.

3. Based on this , we ended up having 3 RCs , one for each zone(see below) since we couldn't find a better way of mapping it. We can still workaround with this for now, but I'd expect EBS volumes to be attached to a different node if a pod goes down  but that doesn't seem to be happening as well.  The error is that  it says that the EBS volume is attached to a different node but in AWS, it clearly shows that the volume is in 'detached' state. I'd really like some help on this issue.
```
apiVersion: v1
kind: ReplicationController
metadata:
  labels:
    app: influxdb
  name: influxdb-peer-2a-rc
  namespace: infra
spec:
  replicas: 1
  selector:
    app: influxdb
    type: peer
  template:
    metadata:
      labels:
        app: influxdb
        type: peer
    spec:
      containers:
      - name: influxdb
        image: supershal/influxdb:latest
        env:
        - name: NAMESPACE
          value: infra
        - name: INFLUXDB_POD_SELECTORS
          value: app=influxdb
        - name: INFLUXDB_META_RAFT_PROMOTION_ENABLED
          value: "false"
        - name: INFLUXDB_POD_IP
          valueFrom:
            fieldRef:
              fieldPath: status.podIP
        - name: INFLUXDB_META_HTTP_BIND_ADDRESS
          value: "$(INFLUXDB_POD_IP):8091"
        - name: INFLUXDB_META_BIND_ADDRESS
          value: "$(INFLUXDB_POD_IP):8088"
        - name: INFLUXDB_HTTP_BIND_ADDRESS
          value: "$(INFLUXDB_POD_IP):8086"
        volumeMounts:
        - mountPath: /data
          name: influxdb-storage
        ports:
        - name: http
          containerPort: 8083
        - name: api
          containerPort: 8086
        - name: meta
          containerPort: 8088
        - name: meta-http
          containerPort: 8091
      volumes:
        - name: influxdb-storage
          persistentVolumeClaim:
            claimName: influxclaim1
      nodeSelector:
        zone: "us-west-2a"
```

**Comment by saad-ali** [comment]
@kubernetes/rh-storage

**Comment by justinsb** [comment]
@madhurranjan can you try again with 1.2, now that we have official multizone support?

http://kubernetes.io/docs/admin/multiple-zones/

We've also fixed a number of bugs around volume mounting (as seen in #2)

**Comment by justinsb** [comment]
Closing as this should now be done with petsets, and likely was fixed by fixes in 1.2 (volume fixes & multizone support)
